### ISSUE TITLE
Style/AlignHash: Align the elements of a hash literal if they span more than one line.

### ISSUE THREAD

**Issue by tmornini** [bbatsov/rubocop#1661]
Hello!

Getting that offense on 0.29.1 where I didn't get them before.

Been able to figure out most of them, but this one has me stumped. :-)

```
                  'Content-Length'                   =>
                    'length',
                  'Content-Type'                     =>
                    'application/json; charset=utf-8',
                  'Access-Control-Max-Age'           =>
                    '1728000',
                  'Access-Control-Allow-Methods'     =>
                    'POST, PUT, GET, PATCH, HEAD, OPTIONS',
                  'Access-Control-Allow-Origin'      =>
                    Constants::BASE_URL,
                  'Access-Control-Allow-Credentials' =>
                    'true'
```

I'm not able to do anything simpler due to line length limits...

**Comment by bbatsov** [comment]
Hmm, I'll have to think about this. I've never had to write a hash in this manner and I feel that breaking it up like this seriously impairs the readability.

**Comment by alexdowad** [comment]
There is a good reason why cops can be disabled.

Sometimes there may be rare situations where a general style rule just doesn't apply. But we can't cater for every such rare situation. A better solution is to simply disable the cop.

**Comment by tmornini** [comment]
@alexdowad I agree 100%, just wanted to show a situation I came across.

**Comment by alexdowad** [comment]
@tmornini Thanks for showing us, and sorry that there doesn't seem to be a better solution right now.

Maintainers, please close.

**Comment by martinverdejo** [comment]
I encountered this today.

I originally had 

> def foo
>   @bar = { en: "Lorem",
>     es: "Ipsum",
>     fr: "Dolor"
>   }
> end


It has to be

> def foo
>   @bar = {
>     en: "Lorem",
>     es: "Ipsum",
>     fr: "Dolor"
>   }
> end

**Comment by bar** [comment]
@martinverdejo me, uhh?

**Comment by bsolomud** [comment]
Elements of a hash can be just aligned literal, like this:
```
def foo
  @bar ||= { end: 'Lorem',
             es: 'Ipsum',
             fr: 'Dolor' }
end
```

**Comment by vongruenigen** [comment]
We have the same problem, that we'd like to align a hash as @akimd did, but rubocop complains. Aligning a hash in such a way is especially useful if the lengths of the keys differ greatly from one another, because it increases readability imo:

Currently required by rubocop:
```ruby
my_hash = {
  abc: 10,
  def: 20,
  ghijklmno: 30,
  p: 40,
  qrstuvwyxzw: 50
}
```

What I'd wish would also be accepted:
```ruby
my_hash = {
  abc:         10,
  def:         20,
  ghijklmno:   30,
  p:           40,
  qrstuvwyxzw: 50
}
```

**Comment by fabriciofreitag** [comment]
why is this being ignored?

**Comment by grimm26** [comment]
> why is this being ignored?

because it was closed 5 years ago :)

**Comment by fabriciofreitag** [comment]
![image](https://user-images.githubusercontent.com/4108128/78694627-53591100-78d3-11ea-9e52-d8e10da708cc.png)

**Comment by bjtucker** [comment]
Just stumbled on this issue in some of our code. Solved it this way to troll a coworker.  rubocop approved. Coworker did not.

```
my_hash = {
  'foo' => '   perfectly_aligned_value_1'.strip,
  'z' => '     perfectly_aligned_value_2'.strip,
  'bar' => '   perfectly_aligned_value_3'.strip,
  'baz' => '   perfectly_aligned_value_4'.strip,
  'bletch' => 'perfectly_aligned_value_5'.strip,
  'dr' => '    perfectly_aligned_value_6'.strip,
}
```

**Comment by bjtucker** [comment]
But more seriously, it looks those of us who showed up since 2019 are concerned about a different issue than the one this ticket was opened for. 
It looks like a more relevant bug about value, rather than key alignment would be here, and perhaps recently resolved?
https://github.com/rubocop-hq/rubocop/issues/7291

**Comment by KelseyDH** [comment]
@bjtucker Looking at https://github.com/rubocop/rubocop/blob/master/spec/rubocop/cop/layout/hash_alignment_spec.rb

I notice there are not specs dealing with values of varying characters.  All of these alignments should be valid:


```
my_hash = {
  abc:         10,
  def:         20,
  ghijklmno:   30,
  p:           40,
  qrstuvwyxzw: 50
}
```

```
my_hash = {
  abc:         1,
  def:         20,
  ghijklmno:   30,
  p:           40,
  qrstuvwyxzw: 50
}
```

```
my_hash = {
  abc:          1,
  def:         20,
  ghijklmno:   30,
  p:           40,
  qrstuvwyxzw: 50
}
```

I haven't  confirmed, but I suspect the one of the latter two examples makes Rubocop unhappy.

**Comment by benbonnet** [comment]
@KelseyDH this is the only one that got accepted on my side (default rubocop, format similar to @bsolomud 's one )

```
        { 'a1' => 1.0 / (24 * 60),
          'a2' => 5.0 / (24 * 60),
          'a23' => 15.0 / (24 * 60),
          'a4' => 1.0 / 24,
          'a433' => 4.0 / 24,
          'a432' => 12 / 24,
          'a12' => 1,
          'ab' => 7 }
```
